### ISSUE TITLE
Magento 2.4.2 support

### ISSUE THREAD

**Issue by romainruaud** [Smile-SA/elasticsuite#2069]
Magento 2.4.2 should be released today. 

We need to ensure that everything is still compatible with this version, and working as expected.

**Comment by KZNcode** [comment]
Got a strange one still diagnosing after update to M2.4.2 last night (our store contains 15,000+ products)... 

A couple of main categories are showing products from completely different parent categories...

**Comment by aart-boer** [comment]
@KZNcode does turning the following setting on fix the problem?

Stores > Config > Catalog > Catalog > Layered Navigation > Display Category filter > Yes

We experienced an issue where every product is being loaded on every category page no matter what filtering is setup under the category. However, displaying category filters under the layered navigation magically fixed this.

After reading your issues I suddenly wondered if it is related to the ES module.

**Comment by vahonc** [comment]
Hello,

With Magento 2.4.1 version we have issue #1925 linked to displaying all products in each category when the category filter is disabled (Stores > Config > Catalog > Catalog > Layered Navigation > Display Category filter > No) but we already added a fix for this and it will be included in the next release of ElasticSuite module.

Best regards

**Comment by art-boer** [comment]
@vahonc Good to hear, thanks for letting me know!

**Comment by KZNcode** [comment]
> @KZNcode does turning the following setting on fix the problem?
> 
> Stores > Config > Catalog > Catalog > Layered Navigation > Display Category filter > Yes

@art-boer Same issue, no matter if it's 'yes' or 'no'

Also changed all categories to "is_anchor" inside the database and that didn't make a difference either. ( #1925#issuecomment-706273046 )

@vahonc   #1925 [commit: 099e3c3; add new filter model for the current category](https://github.com/vahonc/elasticsuite/commit/099e3c397d262ab2f8e9fdb1784b48152ad9fcfc)

This commit didn't fix our issue either... (I applied this over the current release version)

**Comment by refueledinc** [comment]
@romainruaud Is there an ETA on when we can expect M2.4.2 support? Seems people are having several issues related to M2.4.2 and this extension is blocking our update plans.

**Comment by romainruaud** [comment]
Hi @refueledinc 

TL;DR version : very soon, we are in the final "manual" stage of testing everything against 2.4.2.

Details : 

On the "technical" part, I'd say that I trust our Github actions pipeline which is running fine on Magento 2.4.2 and Elasticsuite 2.10.x  branch. For transparency, here is what this pipeline is doing : https://github.com/Smile-SA/elasticsuite/actions/runs/557994738

- install Magento full stack (apache, php, mysql, elasticsearch)
- install Elasticsuite
- install sample data
- setup:di:compile
- do a full reindex
- test autocompletion url
- test category url
- test search result url
- test graphql search endpoint
- test category graphql query
- test product detail graphql endpoint 

So for now it seems that we avoided the classical pattern "lol it's just a minor version of Magento and ...wait why is the di:compile not passing ? holy shit why did they change that much interfaces definition for a minor version".

But of course this is just a first layer of checking. We also have @vahonc investigating on the topic actually to detect anything that could be malfunctioning in the back-office or in the front-office.

This means I don't expect that much work to be necessary (if any) to be compliant with Magento 2.4.2. I'd even bet that's already the case but I undersand that you are waiting a clear GO from our side.

For me it's unclear if the issue raised in this topic by @KZNcode is related to Magento 2.4.2 or something else.

And last but not least, for now we did not have any issues opened by people complaining with 2.4.2 incompatibility, whereas we are generally spammed day-1 (or even before now that Magento allows early access) by issues if there happens to be any.

So we are confident that we'll be able to pull the trigger soon.

Regards

**Comment by groomershop-mt** [comment]
We have done our internat tests and upgraded to Magento 2.4.2 at production - we do not see any (new) issues related with ElasticSuite and Magento 2.4.2.

**Comment by vahonc** [comment]
Hello all,

I've also done all my investigating and tests on both versions of Magento (Comunity and Enterprise Editions) and I don't detect anything that could be malfunctioning in the back-office or in the front-office. All are works as in the previous version of Magento (2.4.1)

BR,
Vadym

**Comment by refueledinc** [comment]
@romainruaud @vahonc Thanks for the updates. I've updated to magento this morning and so far have not noticed any issues specific to the M2.4.2 update.

**Comment by romainruaud** [comment]
Okay, we can conclude that we are compatible with 2.4.2

I close the issue, don't hesitate to open a new one if needed.
